File: src/main/java/dev/shota/decompiler/window/menu/file/Exit.java
package dev.shota.decompiler.window.menu.file;

import dev.shota.decompiler.window.Window;
import dev.shota.decompiler.window.utils.Language;
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.event.KeyEvent;

public class Exit extends JMenuItem implements ActionListener {

    public Exit() {
        super(Language.get("file.exit"), KeyEvent.VK_Q);
        setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_Q, Toolkit.getDefaultToolkit().getMenuShortcutKeyMaskEx()));
        if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.APP_QUIT_HANDLER)) {
            Desktop.getDesktop().setQuitHandler((e, response) -> actionPerformed(null));
            setVisible(false);
        }
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        Window.INSTANCE.dispose();
    }

}

File: src/main/kotlin/dev/shota/decompiler/window/menu/file/items/Exit.kt
package dev.shota.decompiler.window.menu.file.items

import dev.shota.decompiler.window.Window
import dev.shota.decompiler.window.menu.MenuItem
import java.awt.Desktop
import java.awt.event.ActionEvent
import java.awt.event.KeyEvent

class Exit : MenuItem("file.exit", KeyEvent.VK_Q) {

    init {
        if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.APP_QUIT_HANDLER)) {
            Desktop.getDesktop().setQuitHandler { _, _ -> actionPerformed(null) }
            isVisible = false
        }
    }

    override fun actionPerformed(e: ActionEvent?) {
        Window.dispose()
    }

}